src/app/shared/components/course-form/course-form.component.ts

```diff
@@ -1,19 +1,105 @@
 import { Component } from '@angular/core';
 import {
-  FormBuilder, FormGroup
+  AbstractControl,
+  FormArray,
+  FormBuilder, FormGroup,
+  ValidationErrors,
+  ValidatorFn,
+  Validators
 } from '@angular/forms';
+import { Author } from '@app/shared/types/author.model';
+import { IconNames } from '@app/shared/types/icons.model';
 import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
+import { IconName, IconProp } from '@fortawesome/fontawesome-svg-core';
 import { fas } from '@fortawesome/free-solid-svg-icons';
+import { v4 as uuidv4 } from 'uuid';
+
 
 @Component({
   selector: 'app-course-form',
   templateUrl: './course-form.component.html',
   styleUrls: ['./course-form.component.scss'],
 })
-export class CourseFormComponent {
+export class CourseFormComponent{
+  removeIconName:IconNames = IconNames.TrashCan;
+  removeIcon: IconProp | undefined = ['fas',this.removeIconName as IconName];
```

Comment:
I find this a little overkill. It ok to use `fa-icon` directly in the template. However you might want to create your own IconComponent (similar to ButtonComponent)  as the application gets larger and larger.

Reply:
I refactored this. library is added only in the app constructor( once only ) + new Icon component is made